src/resolvers/dockerfile.ts

```diff
@@ -7,10 +7,15 @@ export default class DockerfileResolver implements Resolver {
 	public name = 'Standard Dockerfile'
 
 	private gotDockerfile: boolean = false
+	// Storing the contents of the Dockerfile allows us to
+	// call the hook on it, without traversing the new tar
+	// stream
+	private dockerfileContents: string = ''
 
 	public entry(file: FileInfo): void {
 		if (file.name === 'Dockerfile') {
 			this.gotDockerfile = true
```

Comment:
Could the flag be inferred from `dockerfileContents` not being an empty string? If it could be empty then maybe set it to `null` and infer from not being `null`?